src/app/public/modules/summary-action-bar/summary-action-bar.component.ts

```diff
@@ -137,7 +163,52 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
       }
       this.changeDetector.detectChanges();
     });
+  }
 
-    this.adapterService.setupResizeListener();
+  private setupTabListener(): void {
+    if (!this.observer) {
+      this.observer = this.observerService.create((mutations: MutationRecord[]) => {
+        if ((<HTMLElement>mutations[0].target).attributes.getNamedItem('hidden')) {
+          this.adapterService.revertBodyElementStyles();
+          this.removeResizeListener();
+        } else {
+          setTimeout(() => {
+            this.adapterService.styleBodyElementForActionBar(this.elementRef);
+            this.setupResizeListener();
+          });
```

Comment:
Why is this in a setTimeout? Were there issues with it running within the lifecycle?

Reply:
Ya. The observer fires when the tab that the action bar is in either is shown or hidden. However, if the user is switching from one tab that has an action bar to another that has a different action bar then without the `setTimeout` It is possible that the new one will run its setup before the old one is removed and the removal of the old one will cause the body styles to be removed. 

The timing here is up to Angular so there isn't much we can do about it but the `setTimeout` ensures that the new action bar is always added after the removal of the old one.